lib/ansible/modules/cloud/vmware/vmware_portgroup.py

```diff
@@ -285,36 +319,33 @@ def create_host_port_group(self, host_system, portgroup_name, vlan_id, vswitch_n
                 self.module.fail_json(msg="Failed to add Portgroup due to generic"
                                           " exception : %s" % to_native(generic_exception))
         else:
-            self.changed = False
             # Change portgroup
-            if desired_pgs[0].spec.vlanId != vlan_id:
-                port_group.spec.vlanId = vlan_id
-                self.changed = True
-            if self.check_network_policy_diff(desired_pgs[0].spec.policy, network_policy):
-                port_group.spec.policy = network_policy
+            port_group.spec.name = portgroup_name
+            port_group.spec.vlanId = vlan_id
+            port_group.spec.policy = network_policy
+            port_group.spec.vswitchName = desired_pgs[0].spec.vswitchName
+            if self.check_network_policy_diff(desired_pgs[0].spec.policy, network_policy) or \
+                    desired_pgs[0].spec.vlanId != vlan_id:
                 self.changed = True
-
-            if self.changed:
-                try:
-                    host_system.configManager.networkSystem.UpdatePortGroup(pgName=self.portgroup_name,
-                                                                            portgrp=port_group.spec)
-                    self.changed = True
-                except vim.fault.AlreadyExists as e:
-                    # To match with other vmware_* modules if portgroup
-                    # exists, we proceed, as user may want idempotency
-                    self.changed = False
-                except vim.fault.NotFound as not_found:
-                    self.module.fail_json(msg="Failed to update Portgroup as"
-                                              " vSwitch was not found: %s" % to_native(not_found.msg))
-                except vim.fault.HostConfigFault as host_config_fault:
-                    self.module.fail_json(msg="Failed to update Portgroup due to host"
-                                              " system configuration failure : %s" % to_native(host_config_fault.msg))
-                except vmodl.fault.InvalidArgument as invalid_argument:
-                    self.module.fail_json(msg="Failed to update Portgroup as VLAN id was not"
-                                              " correct as per specifications: %s" % to_native(invalid_argument.msg))
-                except Exception as generic_exception:
-                    self.module.fail_json(msg="Failed to update Portgroup due to generic"
-                                              " exception : %s" % to_native(generic_exception))
+            try:
```

Comment:
This means you are calling the `UpdatePortGroup` every time you run the module? Even if the state has not changed?

Reply:
Yes. I have seen weird behaviour while calling `UpdatePortGroup` API like [this](https://github.com/ansible/ansible/issues/36773#issuecomment-369659095) So I am calling it every time. Any suggestions ?

Reply:
not sure about bug, but i just wonder why you would call the `UpdatePortGroup` if `self.changed == False` (now, you even call it if nothing changed). You are updating something that doesn't need an update?

Reply:
I tried on my setup, not calling update API if nothing is changed. Tweaked code a bit.